GitUpKit/Interface/GIGraphView.m

```diff
@@ -860,112 +860,170 @@ static void _DrawLine(GILine* line, CGContextRef context, CGFloat offset, CGFloa
+      NSString* branchName = localBranch.name;
+      NSRange branchNameRange = NSMakeRange(multilineTitle.length, branchName.length);
+      [multilineTitle appendFormat:@"%@\n", branchName];
+      [boldRanges addObject:[NSValue valueWithRange:branchNameRange]];
+      [darkRanges addObject:boldRanges.lastObject];
     }
-    ++index;
+    
+    [multilineTitle appendString:@"\n"];
   }
-  index = 0;
+  
   for (GCHistoryRemoteBranch* remoteBranch in branch.remoteBranches) {
-    if (index) {
-      [label appendString:@", "];
-    } else if (label.length) {
-      [label appendString:@"\n"];
-    }
-    [label appendString:remoteBranch.name];
-    ++index;
+    NSString* remoteName = remoteBranch.remoteName;
+    NSString* branchName = remoteBranch.branchName;
+    [multilineTitle appendFormat:@"%@/%@\n", remoteName, branchName];
+    NSRange branchNameRange = NSMakeRange(multilineTitle.length - branchName.length - 1, branchName.length); // -1 to exclude '\n'
+    [boldRanges addObject:[NSValue valueWithRange:branchNameRange]];
+    [darkRanges addObject:boldRanges.lastObject];
   }
-  index = 0;
+  
+  if (branch.remoteBranches.count > 0) {
+    [multilineTitle appendString:@"\n"];
+  }
+  
   for (GCHistoryTag* tag in branch.tags) {
-    if (index) {
-      [label appendString:@", "];
-    } else {
-      if (label.length) {
-        [label appendString:@"\n"];
-      }
-      [label appendString:@"["];
-    }
-    [label appendString:tag.name];
-    ++index;
+    NSString* tagName = tag.name;
+    [multilineTitle appendFormat:@"[%@]\n", tagName];
+    NSRange tagNameRange = NSMakeRange(multilineTitle.length - tagName.length - 2, tagName.length); // -2 to exclude char ']' plus '\n'
+    [boldRanges addObject:[NSValue valueWithRange:tagNameRange]];
+    [darkRanges addObject:boldRanges.lastObject];
   }
-  if (index) {
-    [label appendString:@"]"];
+  
+  if (branch.tags.count > 0) {
+    [multilineTitle appendString:@"\n"];
   }
-  if (!label.length) {
-    [label release];
-    return;  // This should only happen if we have a detached HEAD with no other references pointing to the commit
+  
+  if (multilineTitle.length == 0) {
+    [boldRanges release];
+    [darkRanges release];
+    [multilineTitle release];
+    return; // This should only happen if we have a detached HEAD with no other references pointing to the commit
   }
   
-  // Prepare text
+  // Cache attributes
+  static CTFontRef titleFont = NULL;
+  if (titleFont == NULL) {
+    titleFont = CTFontCreateUIFontForLanguage(kCTFontUIFontSystem, 12.0, CFSTR("en-US"));
+  }
+  static CTFontRef boldFont = NULL;
+  if (boldFont == NULL) {
+    boldFont = CTFontCreateUIFontForLanguage(kCTFontUIFontEmphasizedSystem, 12.0, CFSTR("en-US"));
+  }
+  static NSParagraphStyle* style = nil;
+  if (style == nil) {
+    NSMutableParagraphStyle* mutableStyle = [[NSMutableParagraphStyle alloc] init];
+    mutableStyle.lineHeightMultiple = 0.8;
+    style = [mutableStyle copy];
```

Comment:
There's no need to force the use of a `NSParagraphStyle`. Just keep the `NSMutableParagraphStyle` around.